adot-testbed/app/src/test/java/software/amazon/adot/testbed/TraceIdsWithXRayTests.java

```diff
@@ -0,0 +1,190 @@
+    private GenericContainer<?> collector;
+
+    private GenericContainer<?> createAndStartCollector(String configFilePath) throws IOException {
+
+        // Create an environment variable map
+        Map<String, String> envVariables = new HashMap<>();
+        // Set credentials
+        envVariables.put("AWS_REGION", System.getenv("AWS_REGION"));
+        envVariables.put("AWS_ACCESS_KEY_ID", System.getenv("AWS_ACCESS_KEY_ID"));
+        envVariables.put("AWS_SECRET_ACCESS_KEY", System.getenv("AWS_SECRET_ACCESS_KEY"));
+        // Check if AWS_SESSION_TOKEN is not null before adding it
+        if (System.getenv("AWS_SESSION_TOKEN") != null) {
+            envVariables.put("AWS_SESSION_TOKEN", System.getenv("AWS_SESSION_TOKEN"));
+        }
+
+        var collector = new FixedHostPortGenericContainer<>(TEST_IMAGE)
+            .withCopyFileToContainer(MountableFile.forClasspathResource(configFilePath), "/etc/collector/config.yaml")
+            .withFixedExposedPort(4317, 4317, InternetProtocol.TCP)
+            .withLogConsumer(new Slf4jLogConsumer(collectorLogger))
+            .waitingFor(Wait.forLogMessage(".*Everything is ready. Begin running and processing data.*", 1))
+            .withEnv(envVariables)
+            .withCommand("--config", "/etc/collector/config.yaml");
+
+        collector.start();
+        return collector;
+    }
+
+    @Test
+    void testXRayTraceIdSendToXRay() throws Exception {
+        List<String> traceIds = createTraces(true);
+        validateTracesInXRay(traceIds);
+    }
+
+    @Test
+    void testW3CTraceIdSendToXRay() throws Exception {
+        List<String> traceIds = createTraces(false);
+        validateTracesInXRay(traceIds);
+    }
+
+    List<String> createTraces(boolean useXRayIDGenerator) throws Exception {
+        collector = createAndStartCollector("/configurations/config-xrayExporter.yaml");
+
+        OpenTelemetry otel = openTelemetry(useXRayIDGenerator);
+        Tracer tracer = otel.getTracer("adot-trace-test");
+
+        Attributes attributes = Attributes.of(
+            AttributeKey.stringKey("http.method"), "GET",
+            AttributeKey.stringKey("http.url"), "http://localhost:8080/randomEndpoint"
+        );
+
+        int numOfTraces = 5;
+        List<String> traceIds = new ArrayList<String>();
+        for (int count = 0; count < numOfTraces; count++) {
+            Span span = tracer.spanBuilder("trace-id-test")
+                .setSpanKind(SpanKind.SERVER)
+                .setAllAttributes(attributes)
+                .startSpan();
+            //Format trace IDs to XRay format ({16 bytes} --> {1-<4-bytes>-<12-bytes>})
+            String id = new StringBuilder(span.getSpanContext().getTraceId()).insert(8, "-").insert(0, "1-").toString();
+            traceIds.add(id);
+            span.end();
+        }
+
+        // Takes a few seconds for traces to appear in XRay
+        Thread.sleep(20000);
```

Comment:
I would prefer to implement a retry so that this code can run as fast as possible.  Example: https://github.com/aws-observability/aws-otel-test-framework/blob/terraform/adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java#L249

Reply:
Added retry